packages/nx/bin/nx.ts

```diff
@@ -273,12 +259,30 @@ function _getLatestVersionOfNx(): string {
   }
 }
 
-function _supportedPlatform(): boolean {
+function assertSupportedPlatform() {
   try {
     require('../src/native');
-    return true;
-  } catch {
-    return false;
+  } catch (e) {
+    const err = new Error();
+    if (
+      process.platform == 'win32' ||
+      process.platform == 'darwin' ||
+      process.platform == 'linux' ||
+      process.platform == 'freebsd'
+    ) {
+      err.name = 'Missing Platform Dependency';
+      err.message =
+        `The Nx CLI could not find or load the native binary for your supported platform (${process.platform}-${process.arch}). ` +
+        'This likely means that optional dependencies were not installed correctly, or your system is missing some system dependencies.';
+      if (process.env.NX_VERBOSE_LOGGING == 'true') {
+        err.message += `\n\nAdditional error information: \n${e.message}`;
+      }
+    } else {
+      err.name = 'Platform not supported';
+      err.message = `This platform (${process.platform}-${process.arch}) is currently not supported by Nx.`;
+    }
+
+    throw err;
```

Comment:
```suggestion
  } catch (e) {
    const isSupportedPlatform = ['win32', 'darwin', 'linux', 'freebsd'].includes(process.platform); 
    const err = new Error(isSupportedPlatform 
    	? `The Nx CLI could not find or load the native binary for your supported platform (${process.platform}-${process.arch}). ` +
        'This likely means that optional dependencies were not installed correctly, or your system is missing some system dependencies.'
      : `This platform (${process.platform}-${process.arch}) is currently not supported by Nx.`, { 
        cause: e
       });
    err.name = isSupportedPlatform ? 'Missing Platform Dependency' : 'Platform not supported'
    if (process.env.NX_VERBOSE_LOGGING == 'true') {
        err.message += `\n\nAdditional error information: \n${e.message}`;
    }    
    throw err;
```

Reply:
TypeScript doesnt seem to like the `{ cause: e }` object 😮 
<img width="873" alt="image" src="https://github.com/nrwl/nx/assets/4332460/b3d77f84-5e45-4cd3-af16-6f566e08d122">

Reply:
Interesting, it looks like we'd need to bump `lib` to `es2022` from `es2019`.... I personally think that should be fine, but maybe we should check w/ Jason first?

Reply:
Probably not in this pr. This structure changed so that we do the assert in post install too, so throwing the error here isn't applicable anymore.